airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/SnowflakeBeanFactory.kt

```diff
@@ -146,4 +149,62 @@ class SnowflakeBeanFactory {
 
         return HikariDataSource(datasourceConfig)
     }
+
+    @Singleton
+    fun snowflakeStreamingIngestClient(
+        snowflakeConfiguration: SnowflakeConfiguration,
+        snowflakeSqlNameTransformer: SnowflakeSqlNameTransformer,
+        @Named("snowflakePrivateKeyFileName")
+        snowflakePrivateKeyFileName: String = PRIVATE_KEY_FILE_NAME,
+        @Value("\${airbyte.edition}") airbyteEdition: String,
+    ): SnowflakeStreamingIngestClient {
+        val properties = Properties().apply {
+            // Connection properties
+            setProperty("url", "https://${snowflakeConfiguration.host}")
+            setProperty("account", snowflakeConfiguration.host.substringBefore("."))
+            setProperty("user", snowflakeConfiguration.username)
+            setProperty("role", snowflakeConfiguration.role)
+            setProperty("warehouse", snowflakeConfiguration.warehouse)
+            setProperty("database", snowflakeConfiguration.database)
+            setProperty("schema", snowflakeSqlNameTransformer.transform(snowflakeConfiguration.schema))
+            
+            // Authentication
+            when (snowflakeConfiguration.authType) {
```

Comment:
Looks like we may need to extract/commonize some of the logic from the data source config.  We can definitely do that in a separate PR.